pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/GetPartitionMetadataTest.java

```diff
@@ -0,0 +1,289 @@
+                .getPartitionedTopicMetadataAsync(TopicName.get(tp1), true).join();
+        assertTrue(metadata1.isPresent());
+        assertEquals(metadata1.get().partitions, 3);
+
+        // Binary client.
+        final String tp2 = BrokerTestUtil.newUniqueName("persistent://" + DEFAULT_NS + "/tp");
+        clientWitBinaryLookup.getPartitionsForTopic(tp2).join();
+        Optional<PartitionedTopicMetadata> metadata2 = pulsar.getPulsarResources().getNamespaceResources()
+                .getPartitionedTopicResources()
+                .getPartitionedTopicMetadataAsync(TopicName.get(tp2), true).join();
+        assertTrue(metadata2.isPresent());
+        assertEquals(metadata2.get().partitions, 3);
+
+        // Cleanup.
+        admin.topics().deletePartitionedTopic(tp1, false);
+        admin.topics().deletePartitionedTopic(tp2, false);
+    }
+
+    @DataProvider(name = "autoCreationParamsAll")
+    public Object[][] autoCreationParamsAll(){
+        return new Object[][]{
+            // configAllowAutoTopicCreation, paramCreateIfAutoCreationEnabled, isUsingHttpLookup.
+            {true, true, true},
+            {true, true, false},
+            {true, false, true},
+            {true, false, false},
+            {false, true, true},
+            {false, true, false},
+            {false, false, true},
+            {false, false, false}
+        };
+    }
+
+    @Test(dataProvider = "autoCreationParamsAll")
+    public void testGetMetadataIfNonPartitionedTopicExists(boolean configAllowAutoTopicCreation,
+                                                           boolean paramMetadataAutoCreationEnabled,
+                                                            boolean isUsingHttpLookup) throws Exception {
+        conf.setAllowAutoTopicCreationType(TopicType.PARTITIONED);
+        conf.setDefaultNumPartitions(3);
+        conf.setAllowAutoTopicCreation(configAllowAutoTopicCreation);
+        setup();
+        LookupService lookup = getLookupService(isUsingHttpLookup);
+        // Create topic.
+        final String topicNameStr = BrokerTestUtil.newUniqueName("persistent://" + DEFAULT_NS + "/tp");
+        final TopicName topicName = TopicName.get(topicNameStr);
+        admin.topics().createNonPartitionedTopic(topicNameStr);
+        // Verify.
+        PulsarClient client = PulsarClient.builder().serviceUrl(pulsar.getBrokerServiceUrl()).build();
+        PartitionedTopicMetadata response =
+                lookup.getPartitionedTopicMetadata(topicName, paramMetadataAutoCreationEnabled).join();
+        assertEquals(response.partitions, 0);
+        List<String> partitionedTopics = admin.topics().getPartitionedTopicList("public/default");
+        assertFalse(partitionedTopics.contains(topicNameStr));
+        List<String> topicList = admin.topics().getList("public/default");
+        for (int i = 0; i < 3; i++) {
+            assertFalse(topicList.contains(topicName.getPartition(i)));
+        }
```

Comment:
There is no partitioned topic created. Why do we need to check the partitioned topics?

Reply:
> There is no partitioned topic created. Why do we need to check the partitioned topics?

Just avoid the bug generated in the future makes an issue that creates both partitioned topics and non-partitioned topics.